addons/pinniped/config-controller/hack/ytt/default-package-secret.yaml

```diff
@@ -0,0 +1,18 @@
+#@ load("@ytt:yaml", "yaml")
+#@ load("@ytt:data", "data")
+#@overlay/match-child-defaults missing_ok=True
+
+#@ def config():
+---
+infrastructure_provider: #@ data.values.infrastructure_provider
```

Comment:
non blocking: infrastructure_provider can be derived when cluster object is linked, see https://github.com/vmware-tanzu/tanzu-framework/blob/main/addons/controllers/antrea/antreaconfig_util.go#L50

Reply:
Yes!  We chose to do both.  We want a good val for a successful initial deploy, but [also maintain this in our controller](https://github.com/vmware-tanzu/tanzu-framework/pull/1908/files#diff-6381677685668980cbea8d7b2ee426e76393d615dd2058e0e0f13bf77c3fbfbcR201-R208).